### PR TITLE
Add documentation generation for CLI commands from executors

### LINKED ISSUE #27932: AIP-51 - Executor Specific CLI Commands
### Overview
Some Executors have their own first class CLI commands (now that’s hardcoding/coupling!) which setup or modify various components related to that Executor.
### Examples
- **5a**) Celery Executor commands: https://github.com/apache/airflow/blob/27e2101f6ee5567b2843cbccf1dca0b0e7c96186/airflow/cli/cli_parser.py#L1689-L1734
- **5b**) Kubernetes Executor commands: https://github.com/apache/airflow/blob/27e2101f6ee5567b2843cbccf1dca0b0e7c96186/airflow/cli/cli_parser.py#L1754-L1771
- **5c**) Default CLI parser has hardcoded logic for Celery and Kubernetes Executors specifically: https://github.com/apache/airflow/blob/27e2101f6ee5567b2843cbccf1dca0b0e7c96186/airflow/cli/cli_parser.py#L63-L99
### Proposal
Update the BaseExecutor interface with a pluggable mechanism to vend CLI `GroupCommands` and parsers. Executor subclasses would then implement these methods, if applicable, which would then be called to fetch commands and parsers from within Airflow Core cli parser code. We would then migrate the existing Executor CLI code from cli_parser to the respective Executor class.

Pseudo-code example for vending `GroupCommand`s:
```python
# Existing code in cli_parser.py
...
airflow_commands: List[CLICommand] = [
GroupCommand(
        name='dags',
        help='Manage DAGs',
        subcommands=DAGS_COMMANDS,
    ),
    ...
]
# New code to add groups vended by executor classes
executor_cls, _ = ExecutorLoader.import_executor_cls(conf.get('core', 'EXECUTOR'))
airflow_commands.append(executor_cls.get_cli_group_commands())
...
```

### PR DESCRIPTION
**Pull request by potiuk**
The #29055 moved relevant CLI command definition to executors but it automatically removed the command from generated documentation.

Closes: #27932

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Example Celery docs:


image

**Comment by potiuk** [comment]
I will need to add a second one to add links to the providers from Airflow docs - it will have to be done after this one is merged though due to cyclic celery -> airflow ->  kubernetes -> celery  dependency and I have to make sure the inventory for the Celery / Kubernetes providers is pushed to S3 before the Airflow one might succed.

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -727,3 +730,12 @@ def get_cli_commands() -> list[GroupCommand]:
                 subcommands=KUBERNETES_COMMANDS,
             )
         ]
+
+
+def get_parser() -> argparse.ArgumentParser:
```
Comment (o-nikolas) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py]: Should this be something that goes into the BaseExecutor? It looks generic and we'll need it for all executors that have cli commands.

Comment (potiuk) [reply airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py]: Thought about it. Maybe. The small problem with it ist that this is a code that might be installed on earlier version of airflow. BUT it's only executed during sphinx docs build, so yeah, it is pretty safe to extract common code. On it.

Comment (potiuk) [reply airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py]: Done.

**Comment by potiuk** [comment]
That one should be ready - for now it has only "providers" side - I will add separately the "airflow" side to link to the documentation of the CLI commands in providers cc: @eladkal

**Comment by potiuk** [comment]
All green !

**Review by eladkal — APPROVED** [review]
LGTM

File: airflow/providers/celery/executors/celery_executor.py [hunk 2]
```diff
@@ -480,3 +481,11 @@ def get_cli_commands() -> list[GroupCommand]:
...
             ),
         ]
+
+
+def _get_parser() -> argparse.ArgumentParser:
+    """This method is used by Sphinx to generate documentation.
+
+    :meta private:
+    """
+    return CeleryExecutor._get_parser()
```
Comment (o-nikolas) [inline airflow/providers/celery/executors/celery_executor.py:491]: Very nice now abstracted away to the base class!

**Pull request merged**